crates/core/src/api/path.rs

```diff
@@ -251,7 +242,16 @@ impl Segment {
 
 					val.map(|val| Some((x.to_owned(), val)))
 				}
-				Self::Rest(x) => Some(Some((x.to_owned(), segments.to_vec().into()))),
+				Self::Rest(x) => {
+					// TODO: Null byte validity
```

Comment:
Should be handled by axum, URLs are not allowed to contain NULL bytes as per the HTTP spec, but won't hurt to add

Reply:
All the `TODO: Null byte validity.` comments I left as markers for places where we convert between string and strand. I intend to do a follow up PR where we properly figure out how to handle null bytes.